be/src/udf/java/java_udf.cpp

```diff
@@ -0,0 +1,562 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "udf/java/java_udf.h"
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "column/binary_column.h"
+#include "column/column.h"
+#include "column/fixed_length_column.h"
+#include "column/vectorized_fwd.h"
+#include "common/status.h"
+#include "fmt/compile.h"
+#include "fmt/core.h"
+#include "jni.h"
+#include "runtime/primitive_type.h"
+
+#define ADD_NUMBERIC_CLASS(prim_clazz, clazz, sign)                                                           \
+    {                                                                                                         \
+        _class_##prim_clazz = _env->FindClass("java/lang/" #clazz);                                           \
+        CHECK(_class_##prim_clazz);                                                                           \
+        _value_of_##prim_clazz =                                                                              \
+                _env->GetStaticMethodID(_class_##prim_clazz, "valueOf", "(" #sign ")Ljava/lang/" #clazz ";"); \
+        CHECK(_value_of_##prim_clazz) << "Not Found"                                                          \
+                                      << "(" #sign ")Ljava/lang/" #clazz ";";                                 \
+        _val_##prim_clazz = _env->GetMethodID(_class_##prim_clazz, #prim_clazz "Value", "()" #sign);          \
+        CHECK(_val_##prim_clazz);                                                                             \
+    }
+
+#define DEFINE_NEW_BOX(prim_clazz, cxx_type, CLAZZ, CallType)                                    \
+    jobject JVMFunctionHelper::new##CLAZZ(cxx_type value) {                                      \
+        return _env->CallStaticObjectMethod(_class_##prim_clazz, _value_of_##prim_clazz, value); \
+    }                                                                                            \
+    cxx_type JVMFunctionHelper::val##cxx_type(jobject obj) {                                     \
+        return _env->Call##CallType##Method(obj, _val_##prim_clazz);                             \
+    }
+
+namespace starrocks::vectorized {
+
+JVMFunctionHelper& JVMFunctionHelper::getInstance() {
+    static thread_local std::unique_ptr<JVMFunctionHelper> helper;
```

Comment:
in pipeline engine, a expr maybe bounce back and forth across multiple threads. is thread local safe here ?

Reply:
We need to re-call getInstance() every time we call Expr::evalute